From 327d819d8e8161c31da903e8171a89db97862951 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Wed, 16 Jul 2025 12:24:12 +0100
Subject: [PATCH] build: Add ./configure --with-extra="..."

This is intended for downstream packagers to use, to provide extra
information about the version of the downstream package (such as the
RPM ENVR).  This helps when identifying bugs, especially in packges
which have extensive backports (such as the RHEL packages).  This is
the same as the equivalent option in nbdkit.

In Fedora we intend to use this in the spec file:

  ./configure --with-extra='%{name}-%{version}-%{release}'

resulting in an extra version string something like "libnbd-1.23.4-1.fc43".

(cherry picked from commit a04cda6938a9f60b26cb9aa6d55a0b4ef4d0fe76)
---
 README.md    | 13 +++++++++++++
 configure.ac | 15 +++++++++++++++
 2 files changed, 28 insertions(+)

diff --git a/README.md b/README.md
index 0f6bcdd4..385c0e58 100644
--- a/README.md
+++ b/README.md
@@ -163,6 +163,19 @@ ### Download tarballs
 http://libguestfs.org/download/libnbd
 
 
+### Downstream packagers
+
+If you are packaging libnbd, use:
+
+```
+./configure --with-extra='...'
+```
+
+providing extra information about the distribution, and/or
+distro-specific versions.  It helps us with troubleshooting bug
+reports.  (Also, talk to us!)
+
+
 ## Developers
 
 Install the valgrind program and development headers.
diff --git a/configure.ac b/configure.ac
index 40d4f79f..6fc4342e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -18,6 +18,21 @@
 AC_INIT([libnbd],[1.22.2])
 
 AC_CONFIG_MACRO_DIR([m4])
+
+dnl Extra string, a freeform string defined by downstream packagers.
+dnl eg. If you are packaging libnbd for Linux distro X 1.1, you could
+dnl   ./configure --with-extra="X release 1.1"
+AC_ARG_WITH([extra],
+    [AS_HELP_STRING([--with-extra=...],
+                    [extra version information (for use by packagers)])],
+    [LIBNBD_VERSION_EXTRA="$withval"],
+    [LIBNBD_VERSION_EXTRA=]
+)
+AC_DEFINE_UNQUOTED([LIBNBD_VERSION_EXTRA], ["$LIBNBD_VERSION_EXTRA"],
+                   [Extra version information (for use by packagers)])
+
+AC_MSG_NOTICE([libnbd version $PACKAGE_VERSION ($LIBNBD_VERSION_EXTRA)])
+
 m4_ifdef([AC_USE_SYSTEM_EXTENSIONS],[],
          [m4_define([AC_USE_SYSTEM_EXTENSIONS],[])])
 AC_USE_SYSTEM_EXTENSIONS
-- 
2.47.3

